writing_tests.md

```diff
@@ -0,0 +1,16 @@
+_(in fact, this would be part of the new "docs" repo, not a file in this repo.)_
```

Comment:
_(clears his throat to appear serious)_ So, which parts do you feel should be in-lined, @parkerl ?  As I'm suggesting in this PR, we're really nudging pretty hard to go read that section in the central guide.  I'm of the mind that these characteristics (including how some tooling randomizes tests while others don't) are really part and parcel to this general discussion about test suites.